subprojects/publish/src/main/java/org/gradle/api/publish/internal/PublicationInternal.java

```diff
@@ -36,7 +38,11 @@
      */
     PublicationArtifactSet<T> getPublishableArtifacts();
 
-    T addDerivedArtifact(T originalArtifact, File file);
+    void allArtifacts(Action<? super T> action);
```

Comment:
Should we use `allPublishableArtifacts()` to avoid confusion with `MavenArtifact.artifact()`, `getArtifacts()` etc.?

Reply:
Agreed